src/__tests__/actions/testLoginAction.js

```diff
@@ -0,0 +1,46 @@
+import moxios from "moxios";
+import thunk from "redux-thunk";
+import configureMockStore from "redux-mock-store";
+import * as types from "../../redux/action_types";
+import { loginError, loginSuccessful } from "../../redux/actions/loginAction";
+
+const middleware = [thunk];
+const mockstore = configureMockStore(middleware);
+const emailError = "email is required";
+const loginSuccefullMessage = "mike";
+const regURL = "http://localhost:8000/api/users/login";
+
+describe("action", () => {
+  beforeEach(() => moxios.install());
+  afterEach(() => moxios.uninstall());
+
+  it("should dispatch login successful", async done => {
+    moxios.stubRequest(regURL, {
+      status: 200,
+      response: loginSuccefullMessage
+    });
+
+    const store = mockstore({});
```

Comment:
`store` is being used in multiple places. Define it in `beforeEach()` but then clear all the actions in `afterEach()`, figure out how to do this, this is to ensure that each time a new test runs, the store is reset.